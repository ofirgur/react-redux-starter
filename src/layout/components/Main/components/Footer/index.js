import React from 'react';

import FooterContainer from './style';

export default () => {
    return (
        <FooterContainer>
            <div>FOOTER</div>
        </FooterContainer>
    );
};


