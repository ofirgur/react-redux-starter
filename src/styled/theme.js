export default {
    typography: {
        fontFamily: 'Roboto", "Helvetica", "Arial", sans-serif',
        fontSize: '16px',
        mobileFontSize: '10px'
    },
    breakpoint: {
        xs: '0px',
        sm: '600px',
        md: '960px', 
        lg: '1280px',
        xl: '1920px'
    },
    zIndex: {

    },
    direction: {
        ltr: 'ltr',
        rtl: 'rtl'
    },
    background: {
        body: '#263238',
        main: '#607d8b',
        mainHover: '#78909C'
    },
    color: {
        main: '#fff',
        purple: '#651fff',
        red: '#ff7043',
        link: '#00695C',
        hoveredLink: '#004D40'
    },
    padding: {
        main: '16px',
        mobile: '6px'
    },
    margin: {
        main: '16px',
    },
    shadow: {
        main: '0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)'
    }
};