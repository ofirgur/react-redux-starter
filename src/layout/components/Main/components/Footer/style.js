import styled from 'styled-components';

export default styled.footer`
    padding: ${prop => prop.theme.padding.main};
    background: ${prop => prop.theme.background.main};
    grid-column: span 2;
    display: flex;
    justify-content: center;
`;
