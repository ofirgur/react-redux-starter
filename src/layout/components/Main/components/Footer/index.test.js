import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Footer from './index';
import FooterContainer from './style';
import theme from 'styled/theme';


describe('Footer component', () => {

    const wrapper = shallow(<Footer />);  
    
    it('it should mock the context and render correctly', () => {    
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('it should styled correctly', () => {
        const tree = renderer.create(<FooterContainer theme={theme} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

});