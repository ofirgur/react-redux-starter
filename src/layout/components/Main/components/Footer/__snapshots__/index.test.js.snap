// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component it should mock the context and render correctly 1`] = `
<styled.footer>
  <div>
    FOOTER
  </div>
</styled.footer>
`;

exports[`Footer component it should styled correctly 1`] = `
.c0 {
  padding: 16px;
  background: #607d8b;
  grid-column: span 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<footer
  className="c0"
/>
`;
